uasc/secure_channel.go

```diff
@@ -89,10 +89,13 @@ func NewSecureChannel(endpoint string, c *uacp.Conn, cfg *Config) (*SecureChanne
 		cfg.SecurityMode = ua.MessageSecurityModeNone
 	}
 
+	// make a copy of the config to avoid altering the geniune one
```

Comment:
This looks like a bug fix. Where did you run into this?

Reply:
If I recall correctly, when I call NewSecureChannel a second time with the same c.cfg, the c.cfg keeps the fields "SecureChannelID" and the server _(Quick Start Reference Server from OPC UA foundation)_ reject when I try to open the new secure channel. I can can double check that if you're interested.

Reply:
yes, please. It might make sense to do this in any case.

Reply:
I have run the test a second time and It turns out that I can actually reopen the secure channel  with the same __"SecureChannelID"__ in the config __struct__ but when I try to reactivate the __Session__ with the newly created secure channel I get a __StatusBadTCPSecureChannelUnknown (0x807F0000)__.

![image](https://user-images.githubusercontent.com/25567035/68662362-f4b68380-053c-11ea-8485-4d94aaf3c80c.png)
_see first cfg sent and cfg sent on reconnection_

My guess is (based on the following reference) that since the s.cfg.SecureChannelID is not equal to 0, the new SecureChannelID is ignored and so when the next request is sent the wrong SecureChannelID is sent instead.

https://github.com/unknownet/opcua/blob/7348f07a6e76c0b3d339e30d971a2ea3d991342c/uasc/secure_channel.go#L350-L353

Reply:
resolved by #308